src/file/functions.rs

```diff
@@ -0,0 +1,502 @@
+use anyhow::{Context, Result};
+use log::{debug, warn};
+use std::collections::HashMap;
+use std::fs;
+use std::path::{Path, PathBuf};
+use std::process::{Command, Stdio};
+use std::str::FromStr;
+use stdext::function_name;
+
+#[derive(clap::ValueEnum, Debug, Clone, Eq, Hash, PartialEq)]
+#[allow(non_camel_case_types)]
+pub enum Partition {
+    boot,
+    rootA,
+    cert,
+    factory,
+}
+
+impl FromStr for Partition {
+    type Err = anyhow::Error;
+
+    fn from_str(input: &str) -> Result<Partition, Self::Err> {
+        match input {
+            "boot" => Ok(Partition::boot),
+            "rootA" => Ok(Partition::rootA),
+            "cert" => Ok(Partition::cert),
+            "factory" => Ok(Partition::factory),
+            _ => anyhow::bail!("unknown partition: use either boot, rootA, cert or factory"),
+        }
+    }
+}
+
+// ToDo: find a way to use one implementation "FileCopyParams" instead of "FileCopyToParams" and "FileCopyFromParams"
+#[derive(Clone, Debug)]
+pub struct FileCopyToParams {
+    in_file: std::path::PathBuf,
+    partition: Partition,
+    out_file: std::path::PathBuf,
+}
+
+impl FileCopyToParams {
+    pub fn new(
+        in_file: &std::path::Path,
+        partition: Partition,
+        out_file: &std::path::Path,
+    ) -> Self {
+        FileCopyToParams {
+            in_file: in_file.to_path_buf(),
+            partition,
+            out_file: out_file.to_path_buf(),
+        }
+    }
+}
+
+impl FromStr for FileCopyToParams {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let err_msg = "format not matched: in-file-path,out-partition:out-file-path";
+
+        anyhow::ensure!(
+            s.matches(',').count() == 1 && s.matches(':').count() == 1,
+            err_msg
+        );
+
+        let v: Vec<&str> = s.split(&[',', ':']).collect();
+
+        anyhow::ensure!(v.len() == 3, err_msg);
+
+        let in_file = std::path::PathBuf::from(v[0]);
+        let partition = Partition::from_str(v[1])?;
+        let out_file = std::path::PathBuf::from(v[2]);
+
+        anyhow::ensure!(
+            in_file.try_exists().is_ok_and(|v| v),
```

Comment:
Why not `is_ok()`?

Reply:
it only returns the result but not if the file exists.
anyway I renamed v -> exists